### PR TITLE
Adding xsocs Python application

### PR DESCRIPTION
**Pull request by t20100**
This PR adds the [xsocs](https://gitlab.esrf.fr/kmap/xsocs) pure Python package.

This recipe provides both `xsocs-base` and `xsocs` since the later relies on `pyqt` while the former can be used without the GUI part.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by t20100** [comment]
I agree to be listed here as a maintainer.

File: recipes/xsocs/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        - python -c "import sys, xsocs.test; sys.exit(xsocs.test.run_tests(verbosity=3))"
+      requires:
+        - pip
+
+  - name: xsocs
+    build:
+      noarch: python
+    requirements:
+      run:
+        - {{ pin_subpackage('xsocs-base', max_pin="x.x.x.x") }}
```
Comment (carterbox) [inline recipes/xsocs/meta.yaml:54]:
```suggestion
        - {{ pin_subpackage('xsocs-base', exact=True) }}
```

File: recipes/xsocs/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  home: https://gitlab.esrf.fr/kmap/xsocs
+  summary: Xray Strain Orientation Calculation Software
+  license: MIT
+  license_file: LICENSE
+  doc_url: https://kmap.gitlab-pages.esrf.fr/xsocs/
+  dev_url: https://gitlab.esrf.fr/kmap/xsocs
+
+extra:
+  recipe-maintainers:
+    - t20100
```
Comment (carterbox) [inline recipes/xsocs/meta.yaml:74]:
```suggestion
extra:
  recipe-maintainers:
    - t20100
  feedstock-name: xsocs
```

File: recipes/xsocs/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+{% set name = "xsocs" %}
+{% set version = "2022.9.1" %}
+
+package:
+  name: xsocs-base
```
Comment (carterbox) [inline recipes/xsocs/meta.yaml:5]:
```suggestion
  name: xsocs-split
```

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/xsocs/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      run:
+        - {{ pin_subpackage('xsocs-base', exact=True) }}
+        - silx >=1.0.0
+        - pyopengl
+        - pyqt
+    test:
+      imports:
+        - xsocs
+      commands:
+        - xsocs gui --help
```
Comment (carterbox) [inline recipes/xsocs/meta.yaml:62]:
```suggestion
    test:
      imports:
        - xsocs
      commands:
        - xsocs gui --help
        - pip check
      requires:
        - pip
```

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by t20100** [comment]
@carterbox thanks for the review and the improvements!

**Pull request merged**